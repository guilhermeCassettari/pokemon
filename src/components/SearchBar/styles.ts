
import styled from "styled-components/native";
import { getSize } from "../../helper/getSize";


export const Wrapper = styled.View`
    width: 100%;
    height: ${getSize(40)};
    border-radius: ${getSize(20)};
    border: 1px solid #3663AD;
    justify-content: space-between;
    align-items: center;
    padding-top: ${getSize(12)};
    padding-bottom: ${getSize(12)};
    padding-left: ${getSize(14)};
    padding-right: ${getSize(14)};
    flex-direction: row;
    margin-bottom: ${getSize(20)};
`

export const Submit = styled.TouchableOpacity``

export const Icon = styled.Image`
    width: ${getSize(25)};
    height: ${getSize(25)};
    
`

export const Input = styled.TextInput`
    width: ${getSize(220)};
    font-size: ${getSize(14)};
    text-align: left;
    color: #21386E;
    opacity: 0.5;
`

export const Button = styled.TouchableOpacity`
    width: ${getSize(15)};
    height: ${getSize(15)};
    background-color: '#000';
`

