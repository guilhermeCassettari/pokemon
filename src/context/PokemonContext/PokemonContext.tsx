import React, { createContext, Dispatch, ReactNode, SetStateAction, useEffect, useState } from "react";
import { useGetCard } from "../../hooks/useGetCard";
import { PokeCard } from '../../types/PokeCardType'
import { useGetSingleCard } from "../../hooks/useGetSingleCard";
import { Alert } from "react-native";



export interface ContextProps {
  pokeData: PokeCard[] | [],
  loading: boolean,
  pokeSearch: () => void,
  searchValue: string
  setSearchValue: Dispatch<SetStateAction<string>>
  card: PokeCard[],
  searchLoader: boolean,
  fetchMoreData: () => void
}

interface Props {
  children?: ReactNode
}


export const PokemonContext = createContext<ContextProps>({} as ContextProps)

export const PokemonProvider: React.FC<Props> = ({ children }) => {
  const [card, setCard] = useState<PokeCard[]>([])
  const [searchValue, setSearchValue] = useState('')
  const { pokeSearch, singltPokeData, searchLoader, singleCardError } = useGetSingleCard(searchValue)
  const { pokeData, loading, getCardError, fetchMoreData } = useGetCard()

  useEffect(() => {
    if (pokeData) {
      setCard(pokeData)
    }
  }, [pokeData])

  useEffect(() => {
    if (!!singltPokeData.length && !searchLoader) {
      setCard(singltPokeData)
    } else if (!!pokeData.length && searchValue && !loading && !searchLoader) {
      setCard(pokeData)
      Alert.alert(
        "Dont find any Pokemon",
        "Try again more latter",
        [{ text: "OK" }]
      )
    }
  }, [singltPokeData])

  useEffect(() => {
    if (!searchValue.length) {
      setCard(pokeData)
    }
  }, [searchValue])

  useEffect(() => {
    if (getCardError || singleCardError) {
      Alert.alert(
        "Ocurrent a error",
        "Try again more latter",
        [{ text: "OK" }]
      )
    }
  }, [getCardError, singleCardError])

  return (
    <PokemonContext.Provider value={{
      pokeData,
      loading,
      pokeSearch,
      searchLoader,
      searchValue,
      card,
      setSearchValue,
      fetchMoreData
    }}>
      {children}
    </PokemonContext.Provider>
  )
}
