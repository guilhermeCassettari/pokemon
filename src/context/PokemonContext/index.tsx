export { PokemonContext, PokemonProvider } from './PokemonContext'