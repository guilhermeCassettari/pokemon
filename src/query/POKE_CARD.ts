import { gql } from "@apollo/client"

const POKE_CARD = gql(`
query MyQuery($off: Int!) {
  pokemon_v2_pokemon(order_by: {id: asc}, limit: 12, offset: $off) {
    id
    name
    pokemon_v2_pokemonspecy {
      pokemon_v2_pokemoncolor {
        name
      }
    }
    pokemon_v2_pokemontypes {
      pokemon_v2_type {
        name
      }
    }
  }
}

`)

export default POKE_CARD