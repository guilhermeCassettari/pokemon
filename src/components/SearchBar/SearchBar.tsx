import React, { useState } from "react";
import { Text, NativeSyntheticEvent, TextInputChangeEventData } from "react-native";
import { ChangeEvent } from 'react';

import seachIcon from '../../../assets/icon/searchIcon.svg'



import * as S from './styles'

const SearchBar = () => {
    const [value, setValue] = useState('')
    return (
        <S.Wrapper>
            <S.Icon

            />
            <S.Input
                placeholder="Search PokÃ©mon"
                value={value}
                onChange={(e: NativeSyntheticEvent<TextInputChangeEventData>): void=> {                    
                    setValue(e.nativeEvent.text)
                }}
            />
            <S.Button><Text>{'->'}</Text></S.Button>
        </S.Wrapper>
    )
}

export default SearchBar