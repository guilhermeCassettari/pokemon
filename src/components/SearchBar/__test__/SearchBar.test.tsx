import React from 'react'
import { render } from '@testing-library/react-native';
import { PokemonProvider } from '../../../context/PokemonContext';
import SearchBar from '../SearchBar';
import { MockedProvider } from '@apollo/client/testing'
import POKE_CARD from '../../../query/POKE_CARD';

const mocks = [
    {
        request: {
            query: POKE_CARD
        },
        result: {
            data: {
                "pokemon_v2_pokemon": [
                    {
                        "id": 1,
                        "name": "bulbasaur",
                        "pokemon_v2_pokemonspecy": {
                            "pokemon_v2_pokemoncolor": {
                                "name": "green"
                            }
                        },
                        "pokemon_v2_pokemontypes": [
                            {
                                "pokemon_v2_type": {
                                    "name": "grass"
                                }
                            },
                            {
                                "pokemon_v2_type": {
                                    "name": "poison"
                                }
                            }
                        ]
                    },
                    {
                        "id": 2,
                        "name": "ivysaur",
                        "pokemon_v2_pokemonspecy": {
                            "pokemon_v2_pokemoncolor": {
                                "name": "green"
                            }
                        },
                        "pokemon_v2_pokemontypes": [
                            {
                                "pokemon_v2_type": {
                                    "name": "grass"
                                }
                            },
                            {
                                "pokemon_v2_type": {
                                    "name": "poison"
                                }
                            }
                        ]
                    }
                ]
            }
        }
    }
];

describe('SeatchBar', () => {
    it('Should be render correctly', async () => {
        const { toJSON, getByA11yHint } = render(
            <MockedProvider mocks={mocks} addTypename={false}>
                <PokemonProvider>
                    <SearchBar />
                </PokemonProvider>
            </MockedProvider>
        )
        expect(getByA11yHint(/Type here to search pokemons/i)).toBeTruthy()
        expect(getByA11yHint(/Press to search/i)).toBeTruthy()
        expect(toJSON()).toMatchSnapshot()
    })
})

