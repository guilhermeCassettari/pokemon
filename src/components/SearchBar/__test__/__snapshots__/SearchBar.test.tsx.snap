// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeatchBar Should be render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "#3663AD",
        "borderRadius": 42,
        "borderStyle": "solid",
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 85,
        "justifyContent": "space-between",
        "marginBottom": 42,
        "paddingBottom": 25,
        "paddingLeft": 30,
        "paddingRight": 30,
        "paddingTop": 25,
        "width": "100%",
      },
    ]
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "height": 53,
          "width": 53,
        },
      ]
    }
    testID="searchIcon"
  />
  <TextInput
    accessibilityHint="Type here to search pokemons"
    onChange={[Function]}
    onPointerEnter={[Function]}
    style={
      Array [
        Object {
          "color": "#21386E",
          "fontSize": 30,
          "opacity": 0.5,
          "textAlign": "left",
          "width": 471,
        },
      ]
    }
    value=""
  />
  <View
    accessibilityHint="Press to search"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "height": 53,
            "width": 53,
          },
        ]
      }
    />
  </View>
</View>
`;
